### PR TITLE
IN: Authentication, User Context, Protected Routes

### PR DESCRIPTION
**Pull request by Braagaa**
**What it does:**

Allows users to register, login and logout
GET /user/me authenticates user
POST /logout logs out current user by clearing the jwt httpOnly cookie
UserContext is used to keep User object available globally with other handlers (logout, login, register)
PrivateRoute component created to check if user is authenticated at any Route needed

**What to look for:**

Made some slight changed to some components
Issue #31 and issue #11 are relative

**Possible added features**
Splash animation during the fetching of data

### CODE REVIEW

**Comment by dproc96** [comment]
Just tested it on my end, this looks really good and it was working for me!

File: client/src/contexts/user.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React, { useContext, createContext } from "react";
+import useAuth from "../hooks/useAuth";
+
+const UserContext = createContext(null);
+
+export const useUserContext = function () {
+  const user = useContext(UserContext);
+  return user;
+};
```
Comment (aeciorc) [inline client/src/contexts/user.js:9]: Any reason to do this vs export the context itself?

**Review by aeciorc — APPROVED** [review]
Very nicely done! I'm approving, but before merging, make rendering the navbar conditional to the user being logged in, and also take a look at my comment

File: client/src/contexts/user.js (hunk 1 above)
Comment (Braagaa) [reply client/src/contexts/user.js:9]: It just saves you a line of code where you don't have to import useContext and the context object itself. You just import useUserContext and its there for you.

Comment (aeciorc) [reply client/src/contexts/user.js:9]: Gotcha, that makes sense to me

**Review by aeciorc — APPROVED** [review]

**Pull request merged**